### PR TITLE
Feature/new conanfile dependencies

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Feature: New ``conanfile.dependencies`` model, using a dict {requirement: ConanFileInterface} to prepare for Conan 2.0.
Docs: https://github.com/conan-io/docs/pull/2134

#tags: slow

### CODE REVIEW

File: conans/test/integration/configuration/conf/test_conf_from_br.py [hunk 1]
```diff
@@ -34,8 +34,8 @@ def generate(self):
         """)
     client.save({"conanfile.py": consumer,
                  "android": android_profile}, clean_first=True)
-    client.run("install .")
+    client.run("install . -pr:b=default")
```
Comment (lasote) [inline conans/test/integration/configuration/conf/test_conf_from_br.py:37]: Do we need this change? Might be breaking?

File: conans/client/generators/__init__.py [hunk 2]
```diff
@@ -157,6 +156,7 @@ def write_generators(self, conanfile, old_gen_folder, new_gen_folder, output):
                         generator.generate()
                     continue
                 except Exception as e:
+                    print(traceback.format_exc())
```
Comment (lasote) [inline conans/client/generators/__init__.py]: traceback

**Review by lasote — COMMENTED** [review]
Looking good! I don't see any reason to stop this moving forward.

File: conans/test/integration/configuration/conf/test_conf_from_br.py (hunk 1 above)
Comment (memsharded) [reply conans/test/integration/configuration/conf/test_conf_from_br.py:37]: Actually, this was kind of a bug or error. It allowed to bring [conf] to consumers from build requires in the host context (i.e. test requires like gtest), and at least the initial intention was to limit this to build requires in the build context, only for tools like AndroidNDK and similar build tools.

File: conans/client/generators/__init__.py (hunk 2 above)
Comment (memsharded) [reply conans/client/generators/__init__.py]: Removed. This is something to fix elsewhere, when a generator crash, it won't show a useful traceback to diagnose the crash, I think we should improve this in another PR.

Comment (lasote) [reply conans/client/generators/__init__.py]: Totally agree

**Pull request merged**